File: backend/app/schemas/event.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.application import ApplicationStatus
from app.models.event import ApplicationEventType


class ApplicationEventResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    event_type: ApplicationEventType
    from_status: ApplicationStatus | None
    to_status: ApplicationStatus | None
    occurred_at: datetime
    notes: str | None


class ApplicationEventListResponse(BaseModel):
    items: list[ApplicationEventResponse]

File: frontend/src/types/event.ts
import type { ApplicationStatus } from './application'

export type ApplicationEventType = 'CREATED' | 'STATUS_CHANGED'

export type ApplicationEvent = {
  id: number
  event_type: ApplicationEventType
  from_status: ApplicationStatus | null
  to_status: ApplicationStatus | null
  occurred_at: string
  notes: string | null
}

export type ApplicationEventsResponse = {
  items: ApplicationEvent[]
}
